src/snowflake/snowpark/_internal/data_source/datasource_partitioner.py

```diff
@@ -45,6 +45,7 @@ def __init__(
         custom_schema: Optional[Union[str, StructType]] = None,
         predicates: Optional[List[str]] = None,
         session_init_statement: Optional[List[str]] = None,
+        batch_size: Optional[int] = 1,
```

Comment:
shall we give this a bigger default value or do you want to do it in another PR?